src/strategies/gatenet-total-staked/index.ts

```diff
@@ -0,0 +1,242 @@
+import { Multicaller, subgraphRequest } from '../../utils';
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+import { formatUnits } from '@ethersproject/units';
+
+export const author = 'usagar80';
+export const version = '0.0.1';
+
+const UNSTAKE = 'Unstake';
+const STAKE = 'Stake';
+const FEE_DISTRIBUTION = 'Fee distribution';
+const REWARD_ADDED = 'Reward';
+const CLAIM = 'Claim';
+const OTHERS = 'others';
+
+const abi = ['function LOCK_PERIOD() view returns (uint256)'];
+
+const getTransactionType = (transaction) => {
+  switch (transaction.__typename) {
+    case 'Staking':
+      if (transaction.txName === 'FeeDistribution') {
+        return FEE_DISTRIBUTION;
+      } else if (transaction.txName === 'RewardAdded') {
+        return REWARD_ADDED;
+      } else if (transaction.txName === 'Claim') {
+        return CLAIM;
+      }
+      return OTHERS;
+    case 'CompoundDeposit':
+      return STAKE;
+    default:
+      return UNSTAKE;
+  }
+};
+
+//gatenet-total-staked
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const multi = new Multicaller(network, provider, abi, { blockTag });
+  multi.call('lockPeriod', options.address, 'LOCK_PERIOD');
+  const multiResult = await multi.execute();
+  const lockPeriod = Number(multiResult.lockPeriod);
+
+  const result: Record<string, BigNumberish> = {};
+  const args: {
+    where: { sender_in: any };
+  } = {
+    where: {
+      sender_in: addresses
+    }
+  };
+
+  const query = {
+    compoundDeposits: {
+      __args: args,
+      __typename: true,
+      id: true,
+      sender: true,
+      amount: true,
+      shares: true,
+      time: true
+    },
+    compoundWithdraws: {
+      __args: args,
+      __typename: true,
+      id: true,
+      sender: true,
+      amount: true,
+      shares: true,
+      time: true
+    },
+    stakings: {
+      __typename: true,
+      id: true,
+      txName: true,
+      amount: true,
+      user: true,
+      time: true,
+      unStakeIndex: true
+    }
+  };
+  const transactionsList = await subgraphRequest(
+    'https://api.studio.thegraph.com/query/17252/gatenet-cvm/v0.6.1',
+    query
+  );
```

Comment:
Hey, @usagar80 is there a way to pass a block number into this subgraph request? else the voting power may keep changing when people vote

Reply:
Hey @ChaituVR ,
No there is no way to do this. I agree it can very voting power. i.e. today I have 5K Staked so my Quorum would be 5K. But If I stake more 15K then my quorum would be 15K.

Reply:
you tried passing block to args? maybe is there a way to pass timestamp ?

Reply:
Hey @ChaituVR , thanks for your revert!
Yes, we can pass timestamp and it's related comparison in `_arg`. But I don't see any reason for that. May be if you can help me understand more (as an example).

Reply:
Hey @usagar80 so the problem here is we are not fetching staked info based on block number, so when a proposal is open, the user can stake and get the voting power required, This can affect whole proposal voting,

Reply:
@ChaituVR , as per your suggestion here and on Discord I have implement changes to the strategy. Please review and let me know.